react/features/shared-video/components/native/SharedVideo.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+
+import React, { Component } from 'react';
+import { View } from 'react-native';
+
+import { getLocalParticipant } from '../../../base/participants';
+import { connect } from '../../../base/redux';
+import { ASPECT_RATIO_WIDE } from '../../../base/responsive-ui';
+import { setToolboxVisible } from '../../../toolbox/actions';
+
+import VideoManager from './VideoManager';
+import YoutubeVideoManager from './YoutubeVideoManager';
+import styles from './styles';
+
+declare var interfaceConfig: Object;
+
+type Props = {
+
+    /**
+     * The Redux dispatch function.
+     */
+     dispatch: Function,
+
+    /**
+     * Is the video shared by the local user.
+     *
+     * @private
+     */
+    isOwner: boolean,
+
+    /**
+     * True if in landscape mode.
+     *
+     * @private
+     */
+     isWideScreen: boolean,
+
+    /**
+     * The available player width
+     */
+    playerHeight: number,
+
+    /**
+     * The available player width
+     */
+    playerWidth: number,
+
+    /**
+     * The shared video url
+     */
+     videoUrl: string,
+}
+
+/**
+ * Implements a React {@link Component} which represents the large video (a.k.a.
+ * the conference participant who is on the local stage) on Web/React.
+ *
+ * @extends Component
+ */
+class SharedVideo extends Component<Props> {
+    /**
+     * Initializes a new {@code SharedVideo} instance.
+     *
+     * @param {Object} props - The properties.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.setWideScreenMode(props.isWideScreen);
+    }
+
+    /**
+     * Implements React's {@link Component#componentDidUpdate()}.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps: Props) {
+        const { isWideScreen } = this.props;
+
+        if (isWideScreen !== prevProps.isWideScreen) {
+            this.setWideScreenMode(isWideScreen);
+        }
+    }
+
+    /**
+     * Dispatches action to set the visibility of the toolbox, true if not widescreen, false otherwise.
+     *
+     * @param {isWideScreen} isWideScreen - Whether the screen is wide.
+     * @private
+     * @returns {void}
+    */
+    setWideScreenMode(isWideScreen) {
+        this.props.dispatch(setToolboxVisible(!isWideScreen));
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {React$Element}
+     */
+    render() {
+        const {
+            isOwner,
+            playerHeight,
+            playerWidth,
+            videoUrl
+        } = this.props;
+
+        if (!videoUrl) {
+            return null;
+        }
+
+        return (
+            <View
+                pointerEvents = { isOwner ? 'auto' : 'none' }
+                style = { styles.youtubeVideoContainer } >
+                {videoUrl.match(/http/)
```

Comment:
I think we might want to make this a bit more future-proof my turning it into an object where we only populate a key based on the video type:

```
{
    videoUrl: "https://lalala.mp4"
}
```

```
{
    youtubeId: "mjndnkoldj901"
}
````

This way if we add more sources like Vimeo or what have you we should be able to support them easier, WDYT?

Reply:
i think we need to think about this outside the scope of this PR since it most likely involves breaking backwards compatibility with older clients